### DISCUSSION TITLE
How handle enhanced stereochemistry with 1,4 disubstituted cyclohexanes?

### DISCUSSION THREAD

**Discussion by someonetookmyname** [rdkit/rdkit#7174] [Q&A] [answered]
Hi,

I'm trying to understand the best way to handle substructure- and exact- matching of enhanced stereochemistry depictions of "cis or trans" cases and also "cis and trans" cases of 1,4 disubstituted cyclohexanes.

For reference, the simplest depictions of the pure species seem to be:

cis:

`Cl[C@H]1CC[C@@H](Br)CC1 |a:1,4,r,lp:0:3,5:3|`

trans:

`Cl[C@H]1CC[C@H](Br)CC1 |a:1,4,r,lp:0:3,5:3|`

Whereas the "cis or trans" case seems to have several valid possibilities:

```
Cl[C@H]1CC[C@H](Br)CC1 |a:4,o1:1,r,lp:0:3,5:3|
Cl[C@H]1CC[C@H](Br)CC1 |a:1,o1:4,r,lp:0:3,5:3|
Cl[C@H]1CC[C@@H](Br)CC1 |a:4,o1:1,r,lp:0:3,5:3|
Cl[C@H]1CC[C@@H](Br)CC1 |a:1,o1:4,r,lp:0:3,5:3|
```

Also for the "cis and trans" case:

```
Cl[C@H]1CC[C@H](Br)CC1 |a:4,&1:1,r,lp:0:3,5:3|
Cl[C@H]1CC[C@H](Br)CC1 |a:1,&1:4,r,lp:0:3,5:3|
Cl[C@H]1CC[C@@H](Br)CC1 |a:4,&1:1,r,lp:0:3,5:3|
Cl[C@H]1CC[C@@H](Br)CC1 |a:1,&1:4,r,lp:0:3,5:3|
```

By either substructure- or exact- matching, I have not been able to get each "cis or trans" structure to recognize all the other "cis or trans" structures.  And I have the same problem with the "cis and trans" collection.

Is there a solution/approach to this?

I think all my attempts have been pretty standard, so I've left them out for brevity here, but happy to share as useful.

Thanks for any guidance,

Andrew

**Comment by ptosco** [comment, marked as answer]
@someonetookmyname You could use a `MolBundle`:
```
from rdkit import Chem

cis_or_trans = Chem.MolFromSmiles("Cl[C@H]1CC[C@H](Br)CC1 |a:4,o1:1,r,lp:0:3,5:3|")
cis_or_trans
```
![image](https://github.com/rdkit/rdkit/assets/5244385/262204cb-6f33-4dcc-a3bc-d69c69e56257)

```
cis_and_trans = Chem.MolFromSmiles("Cl[C@H]1CC[C@H](Br)CC1 |a:4,&1:1,r,lp:0:3,5:3|")
cis_and_trans
```
![image](https://github.com/rdkit/rdkit/assets/5244385/0365db02-a59a-44e4-baaf-57cd241b23ee)

```
cis_or_trans_bundle = Chem.FixedMolSizeMolBundle()
for mol in Chem.EnumerateStereoisomers.EnumerateStereoisomers(cis_or_trans):
    cis_or_trans_bundle.AddMol(mol)

cis_and_trans_bundle = Chem.FixedMolSizeMolBundle()
for mol in Chem.EnumerateStereoisomers.EnumerateStereoisomers(cis_and_trans):
    cis_and_trans_bundle.AddMol(mol)

cis_or_trans_list = [
    "Cl[C@H]1CC[C@H](Br)CC1 |a:4,o1:1,r,lp:0:3,5:3|",
    "Cl[C@H]1CC[C@H](Br)CC1 |a:1,o1:4,r,lp:0:3,5:3|",
    "Cl[C@H]1CC[C@@H](Br)CC1 |a:4,o1:1,r,lp:0:3,5:3|",
    "Cl[C@H]1CC[C@@H](Br)CC1 |a:1,o1:4,r,lp:0:3,5:3|"
]

for smi in cis_or_trans_list:
    query = Chem.MolFromSmiles(smi)
    assert query
    assert cis_or_trans_bundle.HasSubstructMatch(query, useChirality=True)

cis_and_trans_list = [
    "Cl[C@H]1CC[C@H](Br)CC1 |a:4,&1:1,r,lp:0:3,5:3|",
    "Cl[C@H]1CC[C@H](Br)CC1 |a:1,&1:4,r,lp:0:3,5:3|",
    "Cl[C@H]1CC[C@@H](Br)CC1 |a:4,&1:1,r,lp:0:3,5:3|",
    "Cl[C@H]1CC[C@@H](Br)CC1 |a:1,&1:4,r,lp:0:3,5:3|"
]

for smi in cis_and_trans_list:
    query = Chem.MolFromSmiles(smi)
    assert query
    assert cis_and_trans_bundle.HasSubstructMatch(query, useChirality=True)
```

**Reply by someonetookmyname** [reply]
Thanks very much! .. this looks like exactly what I need
